### PR TITLE
fix(aio): revert resize to HostListener; delete device svc

### PR DESCRIPTION
**Pull request by wardbell**
Angular change detection bug -> no page update on resize.
Reverting to `@HostListener('window:resize', ['$event.target.innerWidth'])` cures it.
Delete DeviceService which no longer serves a purpose.
Adjusted affected AppComponent and LiveExample tests.

To see bug, checkout the 2nd commit from PR #15887 (or any subsequent commit before this PR) and do the following:
- Make browser narrower than 1032 (the trigger width)
- Nav to any guide page
- Note that there are no top menu items
- Make browser wider than 1032.
- Note that there are **_still_ no top menu items.**
- Click _anywhere_ on the page. Now the top menu and side nav  appear ... as they should have after resize.

Set a breakpoint in `AppComponent#onResize` 
```
  onResize(width) {
    this.isSideBySide = width > this.deviceService.sideBySideWidth;
  }
```
This will confirm that the event is caught and propagated via `DeviceService` to the method. The value of `this.isSideBySide` becomes `true`, the template binding should trigger nav panels to appear. But they don't. `ChangeDetectorRef.detectChanges()` and `setTimeout` tricks are fruitless.

I gave up wondering why. Reversion to the following gives the expected behavior.
```
  @HostListener('window:resize', ['$event.target.innerWidth'])
  onResize(width) {
    this.isSideBySide = width > this.sideBySideWidth;
  }
```
Project timeline dictates moving forward and leaving the lingering questions unanswered.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: aio/src/app/embedded/live-example/live-example.component.spec.ts [hunk 1]
```diff
@@ -62,6 +51,8 @@ describe('LiveExampleComponent', () => {
         liveExampleDe.nativeElement.liveExampleContent = liveExampleContent;
 
         fixture.detectChanges();
```
Comment (gkalpak) [inline aio/src/app/embedded/live-example/live-example.component.spec.ts]: Is this necessary?

File: aio/src/app/shared/device.service.ts [hunk 2]
```diff
@@ -1,28 +0,0 @@
...
-    if (window) {
-      window.onresize = () => this.onResize();
-      this.onResize();
-    } else {
-      // when no window, pretend the display is wide.
-      this.onResize(this.sideBySideWidth + 1);
-    }
-  }
-
-  onResize(width?: number) {
```
Comment (gkalpak) [inline aio/src/app/shared/device.service.ts]: Couldn't you use `@HostListener` here? Would that work?

**Review by gkalpak — APPROVED** [review]

**Comment by IgorMinar** [comment]
I reduced the original issue to https://github.com/angular/zone.js/issues/741, but I do agree with @gkalpak that the host listener solution is WAY cleaner.

**Review by IgorMinar — APPROVED** [review]

**Comment by JiaLiPassion** [comment]
@wardbell , @IgorMinar , I just tried angular/aio, I think maybe the problem is angular/aio/package.json still use zone.js@0.8.4, after upgrade to zone.js@0.8.5, the problem is gone. Because in zone.js@0.8.5, window.onProperty is also patched by zone.js with this PR,https://github.com/angular/zone.js/pull/687.

**Comment by gkalpak** [comment]
@JiaLiPassion, angular/zone#687 still doesn't seem to patch `window.onresize` (which is what we care about here).

**Comment by JiaLiPassion** [comment]
@gkalpak , I just tried with the example here like this.

```javascript
<!DOCTYPE html>
<html>
<head>
  <script src="./zone.js"></script>
  <script>
    Zone.current.fork({name: 'resize'}).run(function() {
      window.onresize = function() {
        console.log('Zone name is ', Zone.current.name);
      }
    });
  </script>
</head>
<body>
</body>
</html>>
```

The output is 

```
Zone name is resize
```

And I also tried angular/aio, and debug at DeviceService 

```javascript
export class DeviceService {

  // Show sidenav next to the main doc when display width on current device is greater than this.
  readonly sideBySideWidth = 1032;

  displayWidth = new ReplaySubject<number>(1);

  constructor() {

    if (window) {
      window.onresize = () => this.onResize();
      this.onResize();
    } else {
      // when no window, pretend the display is wide.
      this.onResize(this.sideBySideWidth + 1);
    }
  }
```

window.onresize is patched by zone.js and the Zone.current.name is `angular`

**Comment by gkalpak** [comment]
@JiaLiPassion, you are right. I assumed that `patchOnProperty` would only patch the properties passed to it as a second argument, but it turns out is patches all `on...` properties it can find on the target.

In that case, upgrading to zone.js 0.8.5 should indeed solve the issue.

**Comment by JiaLiPassion** [comment]
@gkalpak , yes, it will loop all properties which starts with `on`, and try to patch them.
I have tried with the commit before this PR, with zone.js 0.8.4, resize not work, after upgrade to 0.8.5, it worked.

Please check it. Thanks!

**Comment by IgorMinar** [comment]
@JiaLiPassion great work! Thank you! We'll still go ahead with merge in this PR which simplifies our code greatly, but it's good to know that the newer version of zones doesn't suffer from this problem.

**Comment by JiaLiPassion** [comment]
@IgorMinar , sure, glad to help!

File: aio/src/app/shared/device.service.ts (hunk 2 above)
Comment (wardbell) [inline aio/src/app/shared/device.service.ts]: No. I tried it. Angular does not activate that decorator on a service. :(  I think it should but that would be a new feature.

**Comment by IgorMinar** [comment]
That's not so easy. Host listeners need to be associated with host element.
Services don't have any element associated with them so I don't know how
this would work. Defaulting to window is not a good option.

**Pull request closed without merging**